README.md

```diff
@@ -22,7 +22,7 @@ Learn about our [design principles](/docs/developer/design-principles.md) and [a
 
 We publish NGINX Kubernetes Gateway releases on GitHub. See our [releases page](https://github.com/nginxinc/nginx-kubernetes-gateway/releases).
 
-The latest release is [0.4.0](https://github.com/nginxinc/kubernetes-ingress/releases/tag/v0.4.0).
+The latest release is [0.5.0](https://github.com/nginxinc/nginx-kubernetes-gateway/releases/tag/v0.5.0).
```

Comment:
Just found this mistake. Should I update this link in all of our release branches?

@sjberman @pleshakov

Reply:
Yep, we should fix that.

Reply:
I don't expect the release branches' readmes being highly visible, so I don't see it is necessary

Reply:
@pleshakov True, but still don't want bad links :) easy enough to fix.

Reply:
🤔

Reply:
That's true, but visiting release branches directly and viewing the README is still an avenue that a user can take (I've done it before), so while it may be minimal, it's also minimal effort and I think worth fixing.